### PR TITLE
Woodpecker improvements

### PR DESCRIPTION
**Pull request by ambroisie**
## Description of changes

Related to #249602.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by emilylange** [comment]
Could you please fix eval? :)

**Comment by ambroisie** [comment]
Oops, yes indeed. That'll have to be converted to a `litteralExpression` then since it references packages.

**Review by Janik-Haag — COMMENTED** [review]
The changes look sensible but we should probably add a test since the modules are growing in complexity.

**Comment by ambroisie** [comment]
> we should probably add a test since the modules are growing in complexity.

Unfortunately, testing the agent itself makes little sense.

And we can't really test the serve AFAIK, as it needs to be provided a token from Gitea in order to register itself. This can't be provided at evaluation time in the tests, as it needs a pre-existing instance running.

File: nixos/modules/services/continuous-integration/woodpecker/agents.nix [hunk 1]
```diff
@@ -35,6 +35,16 @@ let
         '';
       };
 
+      path = lib.mkOption {
```
Comment (ambroisie) [inline nixos/modules/services/continuous-integration/woodpecker/agents.nix:38]: I forgot the part that uses the option when `git add --path`ing.

**Comment by Janik-Haag** [comment]
> And we can't really test the server AFAIK, as it needs to be provided a token from Gitea in order to register itself. This can't be provided at evaluation time in the tests, as it needs a pre-existing instance running.

IIRC this can all be build with api calls but yes it would be a lot of work.

> Unfortunately, testing the agent itself makes little sense.

Can you elaborate on this?

**Comment by ambroisie** [comment]
The agent can't do anything unless it is connected to the server. I'm not even sure if it doesn't just exit with an error, or if it idles and sits there waiting.

**Comment by Janik-Haag** [comment]
Yes but you can put agent and server in to the same test and add them to the module via passthru test.

**Comment by Janik-Haag** [comment]
I'll work on next week if I find time.

**Comment by ambroisie** [comment]
Yes but as I said, I don't see a good way to setup the server in the tests, given the need for a token.

**Review by kip93 — APPROVED** [review]

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Pull request merged**